src/Microsoft.DotNet.Wpf/src/System.Windows.Primitives/Windows/Win32/Foundation/ComScope.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using Windows.Win32.System.Com;
+
+namespace Windows.Win32.Foundation;
+
+/// <summary>
+///  Lifetime management struct for a native COM pointer. Meant to be utilized in a <see langword="using"/> statement
```

Comment:
💡 Consider adding an analyzer for this type which enforces that instances only appear in `using` statements.

Reply:
Would love to, just need time to.